sql/2020/11_Security/cryptominer_share.sql

```diff
@@ -0,0 +1,38 @@
+#standardSQL
+--share of cryptominers
+WITH
+  total AS (
+  SELECT
+    COUNT(0) AS ttl
+  FROM
+    `httparchive.technologies.2020_10_01_desktop`
+  WHERE
+    category = 'Cryptominers'
+    OR category='Cryptominer')
+SELECT
+  app,
+  freq,
+  ROUND((freq/(
+      SELECT
+        ttl
+      FROM
+        total)),2) AS pct
+FROM (
+  SELECT
+    SUM(freq) AS freq,
+    app
+  FROM (
+    SELECT
+      COUNT(0) AS freq,
+      app
+    FROM
+      `httparchive.technologies.2020_08_01_desktop`
+    WHERE
+      category = 'Cryptominers'
+      OR category='Cryptominer'
+    GROUP BY
+      app)
+  GROUP BY
+    app)
```

Comment:
I don't understand why you need this double inner loop?

This seems to give same results:

```suggestion
  SELECT
    COUNT(0) AS freq,
    app
  FROM
    `httparchive.technologies.2020_08_01_desktop`
  WHERE
    category = 'Cryptominers'
    OR category='Cryptominer'
  GROUP BY
    app)
```

I'm also a little cynical on these stats. Are we really saying only 348 sites out of the 5.5 Million desktop sites we crawl have cryptominers on them? That seems very low.

Also why only querying desktop?

Reply:
Updated query; low number of sites can be explained based on several of the key cryptojacking services shutting down, and low financial benefit as compared to some time ago.

Reply:
OK but your history still stills low usage.

Also worth remembering we only look at home pages so an XSS on an inner page (more likely?) means we couldn't pick that up.

Reply:
Since 2018_12_15 (at the time when HTTPArchive started crawling CrUX list I believe), there seems to be a steady decline in the number of cryptomining inclusions (although the number of visited home pages has been increasing). It could also be that people are not removing inclusions and thus are still being counted as using cryptomining (e.g. CoinHive domain is shut down, so the requests for it get NXDOMAIN).
I don't think we can claim that cryptojacking is dead, but several things point in the direction that it's less fruitful and thus less prominent.